frontend/src/Patients/PatientProfileModal.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import './PatientProfileModal.css';
+import ProfileCard from '../MyProfile/ProfileCard';
+import ApptCard from '../MyProfile/ApptCard';
+import ProfileModal from '../MyProfile/ProfileModal';
+import ApptModal from '../MyProfile/ApptModal';
+import api from '../api';
+
+export default function PatientProfileModal({ isOpen, onClose, patientId }) {
+  const [patient, setPatient] = useState(null);
+  const [appointments, setAppointments] = useState([]);
+  const [isProfileModalOpen, setProfileModalOpen] = useState(false);
+  const [isApptModalOpen, setApptModalOpen] = useState(false);
+  const [selectedAppointment, setSelectedAppointment] = useState(null);
+
+  useEffect(() => {
+    if (isOpen) {
+      const fetchPatientInfo = async () => {
+        try {
+          const response = await api.get(`/api/user/patient/${patientId}`, { withCredentials: true });
+          setPatient(response.data);
+
+          const appointmentsResponse = await api.get(`/api/user/patient/${patientId}/appointments`, { withCredentials: true });
+          setAppointments(appointmentsResponse.data);
+        } catch (error) {
+          console.error('Error fetching patient profile:', error.response ? error.response.data : error.message);
+        }
+      };
+
+      fetchPatientInfo();
+    }
+  }, [isOpen, patientId]);
+
+
+  //Edit Profile Command
+  const handleEditPatient = (formData) => {
+    // Implement the edit functionality
+  };
+
+  const handleDeletePatient = async () => {
+    try {
+      await api.delete(`/api/user/patient/${patientId}`, { withCredentials: true });
+      onClose();
+    } catch (error) {
+      console.error('Error deleting patient:', error.response ? error.response.data : error.message);
+    }
+  };
+
+  const handleSubmitAppointment = async (e, appointmentData) => {
```

Comment:
Would it be possible to modularize these helper functions?  Eg. moving adding, editing, and delete handlers are inside Apptmodal and just passing state getter/setters in the props. 

Would make it a bit more readable

Reply:
Alright, I'll try doing that.